pkg/commands/copy_test.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2018 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package commands
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/GoogleContainerTools/kaniko/pkg/dockerfile"
+	"github.com/GoogleContainerTools/kaniko/testutil"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/moby/buildkit/frontend/dockerfile/instructions"
+	"github.com/sirupsen/logrus"
+)
+
+var copyTests = []struct {
+	name           string
+	SourcesAndDest []string
+	expectedDest   []string
+}{
+	{
+		name:           "copy foo into tempCopyExecuteTest/",
+		SourcesAndDest: []string{"foo", "tempCopyExecuteTest/"},
+		expectedDest:   []string{"foo"},
+	},
+	{
+		name:           "copy foo into tempCopyExecuteTest",
+		SourcesAndDest: []string{"foo", "tempCopyExecuteTest"},
+		expectedDest:   []string{"tempCopyExecuteTest"},
+	},
+}
+
+func TestCopyExecuteCmd(t *testing.T) {
+	cfg := &v1.Config{
+		Cmd:        nil,
+		Env:        []string{},
+		WorkingDir: "../../integration/context/",
+	}
+
+	for _, test := range copyTests {
+		t.Run(test.name, func(t *testing.T) {
+			dirList := []string{}
+
+			logrus.Infof("Running test: %v", test.name)
+
+			cmd := CopyCommand{
+				cmd: &instructions.CopyCommand{
+					SourcesAndDest: test.SourcesAndDest,
+				},
+				buildcontext: "../../integration/context/",
+			}
+
+			buildArgs := copySetUpBuildArgs()
+			dest := cfg.WorkingDir + test.SourcesAndDest[len(test.SourcesAndDest)-1]
+
+			os.RemoveAll(dest)
+
+			err := cmd.ExecuteCommand(cfg, buildArgs)
+
+			fi, ferr := os.Open(dest)
+			if ferr != nil {
+				t.Error()
+			}
+			defer fi.Close()
+			fstat, ferr := fi.Stat()
+			if ferr != nil {
+				t.Error()
+			}
+			if fstat.IsDir() {
+				files, ferr := ioutil.ReadDir(dest)
+				if ferr != nil {
+					t.Error()
+				}
+				for _, file := range files {
```

Comment:
@dgageot Wrote this[ temp directory testutility](https://github.com/GoogleContainerTools/skaffold/blob/master/testutil/tmp.go) for skaffold.

The tests become much readable and very easy to test.,  e.g. https://sourcegraph.com/github.com/GoogleContainerTools/skaffold@1409a9b72718099c99b360ba6f19b2a965ff8b55/-/blob/pkg/skaffold/filemon/changes_test.go#L99

I would shameless copy it :) Beauty of open source and give @dgageot credit :)

Note:`kaniko` is under same license as `skaffold` so we shd fine :)

Reply:
I think this is overkill for the two tests I needed to write here, and I'd like my test to read like all the other tests in the project.